homeassistant/components/generic/config_flow.py

```diff
@@ -182,10 +183,14 @@ async def async_test_still(hass, info) -> tuple[dict[str, str], str | None]:
     return {}, f"image/{fmt}"
 
 
+URL_RE = re.compile("//.*:.*@")
+
+
 def slug_url(url) -> str | None:
     """Convert a camera url into a string suitable for a camera name."""
     if not url:
         return None
+    url = URL_RE.sub("//", url)
     url_no_scheme = urlparse(url)._replace(scheme="")
```

Comment:
Untested, but I think it makes sense to do something like `_replace(username=None)._replace(password=None)` instead?

Reply:
Ok, will try it.

Reply:
Let's use `yarl` instead of urlparse. We use that in most other places in Home Assistant that do URL manipulation too.

Reply:
Done a0cf823